server/game/gameSystems/GameSystemLibrary.ts

```diff
@@ -188,8 +192,12 @@ export function playCardFromHand<TContext extends AbilityContext = AbilityContex
     // playType automatically defaults to PlayFromHand
     return new PlayCardSystem(propertyFactory);
 }
-export function payResourceCost<TContext extends AbilityContext = AbilityContext>(propertyFactory: PropsFactory<IPayResourceCostProperties, TContext>): GameSystem<TContext> {
-    return new PayResourceCostSystem<TContext>(propertyFactory);
+export function exhaustResources<TContext extends AbilityContext = AbilityContext>(propertyFactory: PropsFactory<IExhaustResourcesProperties, TContext>): GameSystem<TContext> {
+    return new ExhaustResourcesSystem<TContext>(Object.assign({ ...propertyFactory, isCost: false }));
```

Comment:
Now that we removed the default we can remove the `isCost: false` here and instead we need to do `isCost: true` in payResourceCost so that it has the right behavior.

Also, just noticed that this assignment is being done wrong. Since the `propertyFactory` parameter can be a function sometimes (i.e. when we use the `(context) => ({ ... })` form) this won't always work. We need to use `GameSystem.appendToPropertiesOrPropertyFactory` as you can see in e.g. `moveToDeck`